app/src/main/cpp/skyline/services/codec/IHardwareOpusDecoderManager.h

```diff
@@ -0,0 +1,61 @@
+// SPDX-License-Identifier: MPL-2.0
+// Copyright © 2021 Skyline Team and Contributors (https://github.com/skyline-emu/)
+
+#pragma once
+
+#include <services/serviceman.h>
+
+namespace skyline::service::codec {
+    /**
+     * @brief Struct used to initialize a multi stream decoder
+     * @see opus_multistream_decoder_init()
+     */
+    struct MultiStreamParameters {
+        i32 sampleRate;
+        i32 channelCount;
+        i32 streams; //!< Number of streams
+        i32 coupledStreams; //!< Number of stereo streams
+        std::array<u8, 0x100> mappings; //!< Array of channel mappings
+    };
+    static_assert(sizeof(MultiStreamParameters) == 0x110);
+
+    /**
+     * @brief This service is used to get an instance of IHardwareOpusDecoder
+     * @url https://switchbrew.org/wiki/Audio_services#hwopus
+     */
+    class IHardwareOpusDecoderManager : public BaseService {
+      public:
+        IHardwareOpusDecoderManager(const DeviceState &state, ServiceManager &manager) : BaseService(state, manager) {}
+
+        /**
+         * @brief Returns an IHardwareOpusDecoder object
+         * @url https://switchbrew.org/wiki/Audio_services#OpenHardwareOpusDecoder
+         */
+        Result OpenHardwareOpusDecoder(type::KSession &session, ipc::IpcRequest &request, ipc::IpcResponse &response);
+
+        /**
+         * @brief Returns the required size for the decoder's work buffer
+         * @url https://switchbrew.org/wiki/Audio_services#GetWorkBufferSize
+         */
+        Result GetWorkBufferSize(type::KSession &session, ipc::IpcRequest &request, ipc::IpcResponse &response);
+
+        /**
+         * @brief Returns an IHardwareOpusDecoder object for multi stream
+         * @url https://switchbrew.org/wiki/Audio_services#OpenHardwareOpusDecoderForMultiStream
+         */
+        Result OpenHardwareOpusDecoderForMultiStream(type::KSession &session, ipc::IpcRequest &request, ipc::IpcResponse &response);
+
+        /**
+         * @brief Returns the required size for the multi stream decoder's work buffer.
+         * @url https://switchbrew.org/wiki/Audio_services#GetWorkBufferSizeForMultiStream
+         */
+        Result GetWorkBufferSizeForMultiStream(type::KSession &session, ipc::IpcRequest &request, ipc::IpcResponse &response);
+
+        SERVICE_DECL(
+            SFUNC(0x0, IHardwareOpusDecoderManager, OpenHardwareOpusDecoder),
+            SFUNC(0x1, IHardwareOpusDecoderManager, GetWorkBufferSize),
+            //SFUNC(0x2, IHardwareOpusDecoderManager, OpenHardwareOpusDecoderForMultiStream),
+            //SFUNC(0x3, IHardwareOpusDecoderManager, GetWorkBufferSizeForMultiStream)
```

Comment:
We don't want unimplemented functions being defined since this just clutters things, while it may not apply as much here. It does for a lot of other services, having it done here and not there ends up being really inconsistent.